### PR TITLE
Fragments of Data Cleaning Chapter

### PR DESCRIPTION
**Pull request by tillahoffmann**
This PR adds the first few sections of the Data Cleaning chapter. Specifically,

- `01-intro.md` containing a brief overview and learning objectives.
- `02-reshaping-data.md` discussing long and wide data and transposition.
- `03-tidying-data.md` discussing row and column filtering, summary statistics, grouping, sorting, etc.
- `Makefile` that builds html versions of the markdown documents including execution. We won't need this to present the course, but it's a nice sanity check that the code in the notebooks actually runs.
- `requirements.in` and `requirements.txt` which specified the versions of various python packages to avoid runtime environment drift.
- `.github/workflows/python-markdown.yml` which sets up a python environment and executes the notebooks in CI. I may need a few iterations to get it to build; sorry about the inbox noise.

A few notes:

- I've stuck the files in a folder for the chapter to keep things organized and used `.md` extensions because the content isn't R. I don't know if that messes with the ottr build?
- Thought I'd get your input on the first few sections of the chapter before committing to the remaining sections. All input welcome!
- I've omitted the "What is Tidy Data?" section because it's not Python-specific and already well covered in the existing course.

### CODE REVIEW

File: .github/workflows/python-markdown.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Python Markdown Notebooks
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+  workflow_dispatch:
+
+jobs:
```
Comment (cansavvy) [inline .github/workflows/python-markdown.yml:10]: This is great! Love it.

File: 05-cleaning-data/01-intro.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Cleaning Data
+
+![Data Science Process](https://docs.google.com/presentation/d/114QYFmKuJ2M5E3tlBw8Gwu2xI09tb8gnrKrNkMY9CG4/export/png?pageid=g145cde68be4_0_70)
+
+## Learning Objectives
+
+You have already learned about what tidy data is, why it is important, and how to tidy up a dataset in the main Data Trail course (see [here](https://datatrail-jhu.github.io/DataTrail/cleaning-data.html) for a refresher). After completing this section, you will be able to:
```
Comment (cansavvy) [inline 05-cleaning-data/01-intro.md:7]: This is great! However it won't render properly if it isn't listed in the bookdown.yml and bookdown doesn't do well with subdirectories. Instead you can make separate chapters in the same 05-cleaning-data.Rmd file by creating a new H1 header. 

We can chat about this later today!

**Review by cansavvy — COMMENTED** [review]
Thanks for getting this going @tillahoffmann ! Couple minor comments about file structure that we can chat about later today. 

Sorry I haven't gotten my chapters up yet -- some other deadlines snuck in front of this course. But I hope to get something out this week.

**Comment by tillahoffmann** [comment]
Closed in favor of #9.

**Pull request closed without merging**